src/EventStore.Plugins/Authentication/IAuthenticationProvider.cs

```diff
@@ -13,5 +15,26 @@ public interface IAuthenticationProvider {
 		/// </summary>
 		/// <param name="authenticationRequest"></param>
 		void Authenticate(AuthenticationRequest authenticationRequest);
+
+		/// <summary>
+		///     Return a unique name used to externally identify the authentication provider.
+		/// </summary>
+		string Name { get; }
+
+		/// <summary>
+		///     Get public properties which may be required for the authentication flow.
+		/// </summary>
+		IEnumerable<KeyValuePair<string,string>> GetPublicProperties();
+
+		/// <summary>
+		///     Create any required endpoints.
+		/// </summary>
+		/// <param name="endpointRouteBuilder"></param>
+		void ConfigureEndpoints(IEndpointRouteBuilder endpointRouteBuilder);
+
+		/// <summary>
+		///     Get supported authentication schemes.
+		/// </summary>
+		IEnumerable<string> GetSupportedAuthenticationSchemes();
```

Comment:
```suggestion
		IReadOnlyList<string> GetSupportedAuthenticationSchemes();
```

Reply:
These are not going to change during the plugins lifetime and there is no point in forcing consumers to materialize an array or list if they want to cache it or iterate over it multiple times.